pythclient/calendar_full_intervals.py

```diff
@@ -0,0 +1,1109 @@
+    (datetime.date(2024, 10, 25), "0000-1700"),
+    (datetime.date(2024, 10, 26), None),
+    (datetime.date(2024, 10, 27), "1700-0000"),
+    (datetime.date(2024, 10, 28), "0000-0000"),
+    (datetime.date(2024, 10, 29), "0000-0000"),
+    (datetime.date(2024, 10, 30), "0000-0000"),
+    (datetime.date(2024, 10, 31), "0000-0000"),
+    (datetime.date(2024, 11, 1), "0000-1700"),
+    (datetime.date(2024, 11, 2), None),
+    (datetime.date(2024, 11, 3), "1700-0000"),
+    (datetime.date(2024, 11, 4), "0000-0000"),
+    (datetime.date(2024, 11, 5), "0000-0000"),
+    (datetime.date(2024, 11, 6), "0000-0000"),
+    (datetime.date(2024, 11, 7), "0000-0000"),
+    (datetime.date(2024, 11, 8), "0000-1700"),
+    (datetime.date(2024, 11, 9), None),
+    (datetime.date(2024, 11, 10), "1700-0000"),
+    (datetime.date(2024, 11, 11), "0000-0000"),
+    (datetime.date(2024, 11, 12), "0000-0000"),
+    (datetime.date(2024, 11, 13), "0000-0000"),
+    (datetime.date(2024, 11, 14), "0000-0000"),
+    (datetime.date(2024, 11, 15), "0000-1700"),
+    (datetime.date(2024, 11, 16), None),
+    (datetime.date(2024, 11, 17), "1700-0000"),
+    (datetime.date(2024, 11, 18), "0000-0000"),
+    (datetime.date(2024, 11, 19), "0000-0000"),
+    (datetime.date(2024, 11, 20), "0000-0000"),
+    (datetime.date(2024, 11, 21), "0000-0000"),
+    (datetime.date(2024, 11, 22), "0000-1700"),
+    (datetime.date(2024, 11, 23), None),
+    (datetime.date(2024, 11, 24), "1700-0000"),
+    (datetime.date(2024, 11, 25), "0000-0000"),
+    (datetime.date(2024, 11, 26), "0000-0000"),
+    (datetime.date(2024, 11, 27), "0000-0000"),
+    (datetime.date(2024, 11, 28), "0000-0000"),
+    (datetime.date(2024, 11, 29), "0000-1700"),
+    (datetime.date(2024, 11, 30), None),
+    (datetime.date(2024, 12, 1), "1700-0000"),
+    (datetime.date(2024, 12, 2), "0000-0000"),
+    (datetime.date(2024, 12, 3), "0000-0000"),
+    (datetime.date(2024, 12, 4), "0000-0000"),
+    (datetime.date(2024, 12, 5), "0000-0000"),
+    (datetime.date(2024, 12, 6), "0000-1700"),
+    (datetime.date(2024, 12, 7), None),
+    (datetime.date(2024, 12, 8), "1700-0000"),
+    (datetime.date(2024, 12, 9), "0000-0000"),
+    (datetime.date(2024, 12, 10), "0000-0000"),
+    (datetime.date(2024, 12, 11), "0000-0000"),
+    (datetime.date(2024, 12, 12), "0000-0000"),
+    (datetime.date(2024, 12, 13), "0000-1700"),
+    (datetime.date(2024, 12, 14), None),
+    (datetime.date(2024, 12, 15), "1700-0000"),
+    (datetime.date(2024, 12, 16), "0000-0000"),
+    (datetime.date(2024, 12, 17), "0000-0000"),
+    (datetime.date(2024, 12, 18), "0000-0000"),
+    (datetime.date(2024, 12, 19), "0000-0000"),
+    (datetime.date(2024, 12, 20), "0000-1700"),
+    (datetime.date(2024, 12, 21), None),
+    (datetime.date(2024, 12, 22), "1700-0000"),
+    (datetime.date(2024, 12, 23), "0000-0000"),
+    (datetime.date(2024, 12, 24), "0000-1700"),
+    (datetime.date(2024, 12, 25), "1700-000"),
```

Comment:
according to [Cboe FX](https://fx.cboe.com/faqs/General_FAQs.jsp) a trade date start time is 5pm ET, this logic is inline with 25 Dec 2023 where 24 was a Sunday so for 25 Dec 2024 it's a Wednesday which means using the same logic, market should be closed from Tuesday 24 Dec 5pm to Wed 25 Dec 5pm

Reply:
thank you @ndiakom!